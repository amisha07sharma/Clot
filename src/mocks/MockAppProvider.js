import React, { useReducer } from "react";
import PropTypes from "prop-types";
import { AppContext } from "../context/AppContext";
import appReducer from "../reducers/appReducer";

const initialState = {
  email: "mocked_email@example.com",
  firstName: "John",
  lastName: "Doe",
};

const MockAppProvider = ({ children }) => {
  const [state, dispatch] = useReducer(appReducer, initialState);

  return (
    <AppContext.Provider value={{ state, dispatch }}>
      {children}
    </AppContext.Provider>
  );
};

MockAppProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MockAppProvider;
