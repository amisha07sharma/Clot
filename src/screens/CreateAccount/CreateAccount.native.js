import React, { useState } from "react";
import { TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import useValidation from "../../hooks/useValidation";
import { WHITE } from "../../constants/colors";
import CLOT from "../../constants/messages.json";
import {
  MainContainer,
  HeadingText,
  TextInputContainer,
  TextWrapper,
  TextContainer,
} from "../../styles/common.style";
import CustomButton from "../../components/CustomButton";

const CreateAccount = ({ onContinue, onResetPress }) => {
  const {
    emailValidation,
    passwordValidation,
    firstNameValidation,
    lastNameValidation,
  } = useValidation();
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);
  const [firstName, setFirstName] = useState(null);
  const [lastName, setLastName] = useState(null);

  const handleTextChange = (text) => {
    setEmail(text);
  };
  const handlePasswordChange = (text) => {
    setPassword(text);
  };
  const handleFirstNameChange = (text) => {
    setFirstName(text);
  };
  const handleLastNameChange = (text) => {
    setLastName(text);
  };
  const disableButton = () => {
    return !(
      emailValidation(email) &&
      passwordValidation(password) &&
      firstNameValidation(firstName) &&
      lastNameValidation(lastName)
    );
  };

  return (
    <MainContainer>
      <HeadingText>{CLOT.CREATE_ACCOUNT_SCREEN.CREATE_ACCOUNT}</HeadingText>
      <TextInputContainer
        testID="FirstNameInput"
        onChangeText={handleFirstNameChange}
        placeholder={CLOT.CREATE_ACCOUNT_SCREEN.FIRST_NAME}
        value={firstName}
      />
      <TextInputContainer
        testID="LastNameInput"
        onChangeText={handleLastNameChange}
        placeholder={CLOT.CREATE_ACCOUNT_SCREEN.LAST_NAME}
        value={lastName}
      />
      <TextInputContainer
        testID="EmailInput"
        onChangeText={handleTextChange}
        placeholder={CLOT.CREATE_ACCOUNT_SCREEN.EMAIL_ADDRESS}
        value={email}
      />
      <TextInputContainer
        testID="PasswordInput"
        onChangeText={handlePasswordChange}
        placeholder={CLOT.CREATE_ACCOUNT_SCREEN.PASSWORD}
        value={password}
      />
      <CustomButton
        testID="ContinueButton"
        title={CLOT.CREATE_ACCOUNT_SCREEN.CONTINUE}
        onPress={onContinue}
        color={WHITE}
        disabled={disableButton()}
      />
      <TextWrapper>
        <TextContainer>
          {CLOT.CREATE_ACCOUNT_SCREEN.FORGOT_PASSWORD}
        </TextContainer>
        <TouchableOpacity onPress={onResetPress} testID="Reset">
          <TextContainer bold={true}>
            {CLOT.CREATE_ACCOUNT_SCREEN.RESET}
          </TextContainer>
        </TouchableOpacity>
      </TextWrapper>
    </MainContainer>
  );
};

CreateAccount.propTypes = {
  onContinue: PropTypes.func.isRequired,
  onResetPress: PropTypes.func.isRequired,
};
export default CreateAccount;
