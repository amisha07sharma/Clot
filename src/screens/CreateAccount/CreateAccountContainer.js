import React from "react";
import PropTypes from "prop-types";
import CreateAccount from "./CreateAccount";

const CreateAccountContainer = ({ navigation }) => {
  const onContinue = () => {
    navigation.navigate("BasicDetails");
  };

  const onResetPress = () => {
    navigation.navigate("ForgotPassword");
  };

  return (
    <CreateAccount
      onContinue={onContinue}
      onResetPress={onResetPress}
      testID="CreateAccount"
    />
  );
};

CreateAccountContainer.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
};

export default CreateAccountContainer;
