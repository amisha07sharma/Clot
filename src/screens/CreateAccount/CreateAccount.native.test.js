import React from "react";
import { View as MockView } from "react-native";
import { render, fireEvent, screen } from "@testing-library/react-native";
import MockAppProvider from "../../mocks/MockAppProvider";
import CreateAccount from "./CreateAccount";

jest.mock("../../hooks", () => ({
  useValidation: () => ({
    passwordValidation: jest.fn(),
    emailValidation: jest.fn(),
    firstNameValidation: jest.fn(),
    lastNameValidation: jest.fn(),
  }),
}));

jest.mock("../../components/CustomButton", () => (props) => (
  <MockView {...props} />
));

const mockOnContinue = jest.fn();
const mockOnResetPress = jest.fn();
const mockProps = {
  onContinue: mockOnContinue,
  onResetPress: mockOnResetPress,
};

const renderComponent = (props = {}) => {
  return render(
    <MockAppProvider>
      <CreateAccount {...mockProps} {...props} />
    </MockAppProvider>
  );
};
describe("Tests for CreateAccount Screen", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should match the snapshot", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("updates email state when text input changes", () => {
    renderComponent();

    const emailInput = screen.getByTestId("EmailInput");

    fireEvent.changeText(emailInput, "new_email@example.com");
    expect(emailInput.props.value).toBe("new_email@example.com");
  });

  it("calls onContinue function when Continue button is pressed", () => {
    renderComponent();

    fireEvent(screen.getByTestId("ContinueButton"), "onPress");

    expect(mockOnContinue).toHaveBeenCalledTimes(1);
  });

  it("calls onResetPress callback when Reset text is pressed", () => {
    renderComponent();

    fireEvent.press(screen.getByTestId("Reset"));

    expect(mockOnResetPress).toHaveBeenCalledTimes(1);
  });

  it("updates state when input fields are changed", () => {
    const { getByTestId } = renderComponent();

    fireEvent.changeText(getByTestId("FirstNameInput"), "name1");
    fireEvent.changeText(getByTestId("LastNameInput"), "lastName");
    fireEvent.changeText(getByTestId("EmailInput"), "email@example.com");
    fireEvent.changeText(getByTestId("PasswordInput"), "password123");
    expect(getByTestId("FirstNameInput").props.value).toBe("name1");
    expect(getByTestId("LastNameInput").props.value).toBe("lastName");
    expect(getByTestId("EmailInput").props.value).toBe("email@example.com");
    expect(getByTestId("PasswordInput").props.value).toBe("password123");
  });

  it("disables Continue button when any input is invalid", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("ContinueButton").props.disabled).toBe(true);

    fireEvent.changeText(getByTestId("FirstNameInput"), "name1");
    fireEvent.changeText(getByTestId("LastNameInput"), "lastName");
    fireEvent.changeText(getByTestId("EmailInput"), "email@example.com");
    fireEvent.changeText(getByTestId("PasswordInput"), "password123");

    expect(getByTestId("ContinueButton").props.disabled).toBe(false);
  });
});
