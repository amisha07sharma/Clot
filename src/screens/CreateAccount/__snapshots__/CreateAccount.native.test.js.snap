// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CreateAccount Screen should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 50,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 32,
        "fontWeight": "700",
        "letterSpacing": -0.41,
        "lineHeight": 34.5,
        "marginBottom": 30,
      }
    }
  >
    Create Account
  </Text>
  <TextInput
    backgroundColor="#f4f4f4"
    borderRadius={4}
    borderWidth={0.1}
    fontSize={16}
    height={56}
    letterSpacing={-0.41}
    lineHeight={20.24}
    marginBottom={20}
    onChangeText={[Function]}
    paddingLeft={10}
    placeholder="FirstName"
    style={
      {
        "fontWeight": "450px",
      }
    }
    testID="FirstNameInput"
    theme={{}}
    value={null}
  />
  <TextInput
    backgroundColor="#f4f4f4"
    borderRadius={4}
    borderWidth={0.1}
    fontSize={16}
    height={56}
    letterSpacing={-0.41}
    lineHeight={20.24}
    marginBottom={20}
    onChangeText={[Function]}
    paddingLeft={10}
    placeholder="LastName"
    style={
      {
        "fontWeight": "450px",
      }
    }
    testID="LastNameInput"
    theme={{}}
    value={null}
  />
  <TextInput
    backgroundColor="#f4f4f4"
    borderRadius={4}
    borderWidth={0.1}
    fontSize={16}
    height={56}
    letterSpacing={-0.41}
    lineHeight={20.24}
    marginBottom={20}
    onChangeText={[Function]}
    paddingLeft={10}
    placeholder="Email Address"
    style={
      {
        "fontWeight": "450px",
      }
    }
    testID="EmailInput"
    theme={{}}
    value={null}
  />
  <TextInput
    backgroundColor="#f4f4f4"
    borderRadius={4}
    borderWidth={0.1}
    fontSize={16}
    height={56}
    letterSpacing={-0.41}
    lineHeight={20.24}
    marginBottom={20}
    onChangeText={[Function]}
    paddingLeft={10}
    placeholder="Password"
    style={
      {
        "fontWeight": "450px",
      }
    }
    testID="PasswordInput"
    theme={{}}
    value={null}
  />
  <View
    color="white"
    disabled={true}
    onPress={[MockFunction]}
    testID="ContinueButton"
    title="Continue"
  />
  <View
    style={
      {
        "flexDirection": "row",
        "marginBottom": 80,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 12,
          "fontWeight": "450",
          "letterSpacing": -0.41,
          "lineHeight": 15.18,
        }
      }
    >
      Forgot Password ?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Reset"
    >
      <Text
        bold={true}
        style={
          {
            "fontSize": 12,
            "fontWeight": "700",
            "letterSpacing": -0.41,
            "lineHeight": 15.18,
          }
        }
      >
         Reset
      </Text>
    </View>
  </View>
</View>
`;
